### PR TITLE
Bugfix and enhancements

### PR DESCRIPTION
**Pull request by andre8244**
- [x] Data migration from ntopng 3
- [x] Fix web interface port in API
- [x] Fix on hosts tables
- [x] Enhance traffic by protocol chart
- [x] Minor UI enhancements

NethServer/dev#6199

### CODE REVIEW

File: root/etc/e-smith/templates/etc/ntopng/ntopng.conf/10base [hunk 1]
```diff
@@ -22,7 +22,7 @@
     }
 
     if (scalar(@interfaces) > 1) {
-        $OUT.="-i view: $ntopng{'Interfaces'}\n";
+        $OUT.="-i view:all\n";
```
Comment (gsanchietti) [inline root/etc/e-smith/templates/etc/ntopng/ntopng.conf/10base]: Remove the `if` is not needed anymore:
```
$OUT.="-i view:all\n";
```

File: root/etc/e-smith/events/actions/nethserver-ntopng-conf [hunk 2]
```diff
@@ -37,3 +37,6 @@ if ( system("/usr/bin/systemctl is-enabled --quiet redis &>/dev/null") == 0 &&
     system("/usr/bin/systemctl is-active --quiet redis &>/dev/null") > 0 ) {
     system("/usr/bin/systemctl disable redis");
 }
+
+# Migrate data from ntopng 3, if any
+system("/etc/e-smith/events/actions/nethserver-ntopng-3-migrate");
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-ntopng-conf]: Remove this code and link the new action inside the `createlinks` script

File: root/etc/e-smith/events/actions/nethserver-ntopng-3-migrate [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#
+
+# Migrate data from ntopng 3, if any
+
+if [ -d /var/tmp/ntopng ]; then
+  systemctl stop ntopng
+  rm -rf /var/lib/ntopng/
+  mv /var/tmp/ntopng /var/lib/ntopng/
+  chown -R ntopng:ntopng /var/lib/ntopng/
+  systemctl start ntopng
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-ntopng-3-migrate]: This line will start ntopng even if it's disabled.
Before starting, check if is enabled.

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: createlinks [hunk 4]
```diff
@@ -36,6 +36,7 @@ event_templates("nethserver-ntopng-update", qw(
 event_actions("nethserver-ntopng-update", qw(
     initialize-default-databases 00
     nethserver-ntopng-conf 02
+    nethserver-ntopng-3-migrate 04
```
Comment (gsanchietti) [inline createlinks]: Positions between 0 and 5 are reserved.
Please move to 20

**Review by gsanchietti — CHANGES_REQUESTED** [review]

**Review by gsanchietti — APPROVED** [review]
:+1:

**Review by edospadoni — APPROVED** [review]

**Pull request merged**